### PR TITLE
UPDATED perf: optimize performance of waybar cpuinfo and gpuinfo modules

### PR DESCRIPTION
**Pull request by boulder-roller**
# Pull Request

## 
This is a copy of #952 but updated so that it could be auto-merged

## Checklist

Please put an `x` in the boxes that apply:

- [ ] I have read the [CONTRIBUTING](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My commit message follows the [commit guidelines](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added a changelog entry.
- [ ] I have added necessary comments/documentation to my code.
- [ ] I have added tests to cover my changes.
- [x] I have tested my code locally and it works as expected.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by T-Crypt** [comment]
I need to review the PR from Feb / March before merging

**Review by T-Crypt — APPROVED** [review]
Approving and testing

**Comment by mislah** [comment]
Thanks for working on updating the code and resolving the merge conflicts. While this pull request is based on our earlier work from #952, and most of the code remains unchanged, I noticed that none of the commits from my original pull request are included here.  

As someone who dedicated valuable time to this contribution, I believe it's basic etiquette to preserve the proper authorship . I request the maintainers @T-Crypt and @prasanthrangan to revert this PR and consider opening a new one that retains both my and @kRHYME7's commit history and authorship.

**Pull request merged**
